### PR TITLE
Upload YOLOv5 Models to Roboflow Deploy

### PR DESCRIPTION
**Pull request by Jacobsolawetz**
# Description

Adds ability to parse YOLOv5 models to Roboflow deploy upload

## Type of change

-   [x ] New feature (non-breaking change which adds functionality)

## How has this change been tested, please provide a testcase or example of how you tested the change?

I have tested it via CI and locally, and in notebook

## Docs

-   [ n] Docs updated? What were the changes:

### CODE REVIEW

**Review by paulguerrie — APPROVED** [review]
LGTM!  Tested while pairing with Jacob on the conversion server.

File: roboflow/core/version.py [hunk 1]
```diff
@@ -300,24 +300,32 @@ def deploy(self, model_type: str, model_path: str) -> None:
...
-            class_names = []
-            for i, val in enumerate(model["model"].names):
-                class_names.append((val, model["model"].names[val]))
-            class_names.sort(key=lambda x: x[0])
-            class_names = [x[1] for x in class_names]
+        class_names = []
+        for i, val in enumerate(model["model"].names):
+            class_names.append((val, model["model"].names[val]))
+        class_names.sort(key=lambda x: x[0])
+        class_names = [x[1] for x in class_names]
```
Comment (FrancescoSaverioZuppichini) [inline roboflow/core/version.py:326]: I know this is not for this PR, but could we add comments about what is going on for yolov8?

**Review by FrancescoSaverioZuppichini — APPROVED** [review]
LGTM! I can refactor the code in a follow up PR, we have too many conditions that can be fix by dep injection, e.g.

```python
def deploy_yolov5(...):
   ....
   
def deploy_yolov8(...):
  ...

def deploy(..., model_format):
   if model_format == "yolov5":
      deploy_yolov5(...)
 ....
```

**Comment by Jacobsolawetz** [comment]
@FrancescoSaverioZuppichini that would be awesome to parcel this off and to get ready for more models 🚀

File: roboflow/core/version.py [hunk 2]
```diff
@@ -159,7 +159,7 @@ def download(self, model_format=None, location=None, overwrite: bool = True):
 
         if model_format == "yolov8":
             # we assume the user will want to use yolov8, for now we only support ultralytics=="8.11.0"
-            print_warn_for_wrong_dependencies_versions(["ultralytics", "<=", "8.0.20"])
+            print_warn_for_wrong_dependencies_versions([["ultralytics", "<=", "8.0.20"]])
```
Comment (FrancescoSaverioZuppichini) [inline roboflow/core/version.py:162]:
```suggestion
print_warn_for_wrong_dependencies_versions([("ultralytics", "<=", "8.0.20")])
```

**Pull request merged**